hwbench/environment/memory.py

```diff
@@ -0,0 +1,56 @@
+import json
+import subprocess
+from typing import Any
+
+
+class KernelMemoryInfo:
+    def __init__(self) -> None:
+        self._raw: dict[str, int] = {}
+
+    def detect(self):
+        try:
+            with open("/proc/meminfo") as f:
+                for line in f:
+                    key, value = line.split(":")
+                    try:
+                        amount, unit = value.strip().split()
+                    except ValueError:
+                        amount = value.strip()
+                        unit = ""
+                    key_snake = self._to_snake_case(key)
+                    self._raw[key_snake] = self._convert_to_bytes(int(amount), unit)
+        except FileNotFoundError:
+            raise RuntimeError("/proc/meminfo not found. Are you running on Linux?")
+
+    def dump(self):
+        return dict(self._raw)
+
+    def _to_snake_case(self, name):
+        return name.replace("(", "").replace(")", "").replace("-", "_").strip().lower()
```

Comment:
This function seems useless:
 - no need to remove ( or )
 - there are no - in meminfo
 - it does not really convert CamelCase to snake_case